src/uct/tcp/tcp_cm.c

```diff
@@ -435,9 +497,11 @@ uct_tcp_cm_handle_conn_req(uct_tcp_ep_t **ep_p,
     ucs_assertv(!(ep->flags & UCT_TCP_EP_FLAG_CTX_TYPE_TX),
                 "ep %p mustn't have TX cap", ep);
 
-    if (!uct_tcp_ep_is_self(ep) &&
-        (peer_ep = uct_tcp_cm_search_ep(iface, &ep->peer_addr,
-                                        UCT_TCP_EP_FLAG_CTX_TYPE_TX))) {
+    connect_to_self = uct_tcp_ep_is_self(ep);
+    if (!connect_to_self &&
+        ((peer_ep = uct_tcp_cm_get_ep(iface, &ep->peer_addr,
+                                      cm_req_pkt->conn_sn,
+                                      UCT_TCP_EP_FLAG_CTX_TYPE_TX)) != NULL)) {
```

Comment:
if (!connect_to_self) {
   peer_ep = ... 
   if (peer_ep == NULL) goto out;
   ...
}

Reply:
can we replace 'else' by 'goto out'?
seem ugly to call function with many arguments inside the if condition